### PR TITLE
Fix(frontend/backend) improve performances on space or user list in administration page

### PR DESCRIPTION
**Pull request by Haskaris**
See #3933

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

Here is the methods I used to create 100 spaces and 100 users:
```
makeRandomString (length = 5) {
  var text = ''
  var possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
  
  for (var i = 0; i < length; i++) {
    text += possible.charAt(Math.floor(Math.random() * possible.length))
  }
  
  return text
}
    
randomUser () {
  const fullName = this.makeRandomString()
  const username = this.makeRandomString()
  
  const data = {
    email: `${fullName}@tracim.fr`,
    email_notification: false,
    lang: 'en',
    password: '8QLa$a',
    profile: 'users',
    public_name: fullName,
    username: username,
    timezone: 'Europe/Paris'
  }
  
  return data
}

randomWorkspace () {
  const workspaceName = this.makeRandomString()

  const data = {
    description: `A super description of ${workspaceName}.`,
    label: workspaceName,
    access_type: 'confidential',
    default_user_role: 'reader'
  }

  return data
}
    
async pushRandomUser () {
  for (let i = 0; i < 100; i++) {
    const user = this.randomUser()
    await postAddUser(this.state.config.apiUrl, user.public_name, user.username, user.email, user.profile, user.email_notification, user.password)
  }
}

async pushRandomWorkspace () {
  for (let i = 0; i < 100; i++) {
    const ws = this.randomWorkspace()
    await baseFetch('POST', `${this.state.config.apiUrl}/workspaces`, {
      access_type: ws.access_type,
      default_user_role: ws.default_user_role,
      label: ws.label,
      description: ws.description
    })
  }
}
```
If you use those, don't forget to add `baseFetch` in import from `tracim_frontend_lib`

- [x] Manual, quality tests have been done

### CODE REVIEW

**Review by inkhey — CHANGES_REQUESTED** [review]
Can you update some api test to check theses endpoint in backend ?

File: frontend_app_admin_workspace_user/src/container/AdminWorkspaceUser.jsx [hunk 1]
```diff
@@ -192,22 +180,13 @@ export class AdminWorkspaceUser extends React.Component {
     return fetchUserDetail.body
   }
 
-  displayErrorFetchingUserList () {
-    sendGlobalFlashMessage(this.props.t('Error while loading users list'))
-  }
-
```
Comment (gbonaspetti) [inline frontend_app_admin_workspace_user/src/container/AdminWorkspaceUser.jsx:198]: getDetailedUser uses this function too

**Review by gbonaspetti — CHANGES_REQUESTED** [review]

File: frontend_app_admin_workspace_user/src/container/AdminWorkspaceUser.jsx (hunk 1 above)
Comment (Haskaris) [reply frontend_app_admin_workspace_user/src/container/AdminWorkspaceUser.jsx:198]: Thanks!

**Comment by Haskaris** [comment]
I've converted this PR to draft since my changes are involving more changes than I thought

**Review by inkhey — APPROVED** [review]
:+1:

File: frontend_app_admin_workspace_user/test/container/AdminWorkspaceUser.spec.js [hunk 2]
```diff
@@ -279,6 +279,7 @@ describe('<AdminWorkspaceUser />', () => {
 
         it('should replace the modified workspace', () => {
           const workspaceList = wrapper.state('content').workspaceList
+          console.log(workspaceList)
```
Comment (raphj) [inline frontend_app_admin_workspace_user/test/container/AdminWorkspaceUser.spec.js:282]: Leftover?

File: frontend_app_admin_workspace_user/test/setup.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ import chai from 'chai'
 import Enzyme from 'enzyme'
 import chaiEnzyme from 'chai-enzyme'
 import Adapter from 'enzyme-adapter-react-16'
-// import sinon from 'sinon'
+import sinon from 'sinon'
```
Comment (raphj) [inline frontend_app_admin_workspace_user/test/setup.js:6]: Leftover?

File: frontend_app_admin_workspace_user/test/setup.js [hunk 4]
```diff
@@ -32,4 +32,4 @@ if (!global.window && !global.document) {
 
 Enzyme.configure({ adapter: new Adapter() })
 chai.use(chaiEnzyme())
-// sinon.stub(console, 'log')
+sinon.stub(console, 'log')
```
Comment (raphj) [inline frontend_app_admin_workspace_user/test/setup.js:35]: Leftover?

**Review by gbonaspetti — APPROVED** [review]

**Comment by gbonaspetti** [comment]
Tested on branch

**Pull request merged**